dream_layer_backend/dream_layer_backend_utils/batch_report_generator.py

```
@@ -0,0 +1,364 @@
+    
+    def _create_zip(self, source_dir: str, zip_path: str) -> None:
+        """
+        Create ZIP file from the temporary directory.
+        
+        Args:
+            source_dir: Directory containing files to zip
+            zip_path: Path where the ZIP file will be created
+        """
+        try:
+            with zipfile.ZipFile(zip_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
+                for root, dirs, files in os.walk(source_dir):
```

Ruff (0.12.2)

Loop control variable `dirs` not used within loop body

Rename unused `dirs` to `_dirs`

(B007)